serverAkkaHttp/src/main/scala/RouterDerivation.scala

```diff
@@ -16,26 +16,25 @@ trait RouterDerivationModule extends PathMacro with MetaDataMacro with TypePathM
 
 object RouterDerivationMacro extends RouterDerivationModule {
   //val is required to make universe public
-  class MacroHelper[U <: Universe](val universe: U) {
-    import universe._
-
+  object MacroHelper {
     //check only annotations of path type
     //Since universe is public Tree type can be returned
-    def derivePath(tpe: Type): Tree = tpe.typeSymbol.annotations.collectFirst {
-      case pathAnnotation if pathAnnotation.tree.tpe <:< weakTypeOf[path] =>
-        q"override val path = derivePath[$tpe]"
-    }.getOrElse(EmptyTree)
+    def derivePath[U <: Universe](universe: U)(tpe: universe.Type): universe.Tree = {
+      import universe._
+      tpe.typeSymbol.annotations.collectFirst {
+        case pathAnnotation if pathAnnotation.tree.tpe <:< weakTypeOf[path] =>
+          q"override val path = derivePath[$tpe]"
+      }.getOrElse(EmptyTree)
+    }
   }
 
-  def deriveRouterImpl[A: c.WeakTypeTag](c: Context)(a: c.Expr[A]): c.Tree = {
+  def deriveRouterImpl[A: c.WeakTypeTag](c: Context)(a: c.Tree): c.Tree = {
     import c.universe._
     val tpe = weakTypeOf[A]
-    val derivePath = new MacroHelper[c.universe.type](c.universe).derivePath(tpe)
+    val derivePath = MacroHelper.derivePath(c.universe)(tpe)
 
     q"""
-    import wiro.{ OperationType, MethodMetaData }
```

Comment:
@calippo I removed this imports and the tests still pass. Is there a reason why these were here or is it a leftover from a previous implementation?

Reply:
This shouldn't compile

![image](https://user-images.githubusercontent.com/2476080/35513167-74ec2aa6-0502-11e8-9b79-0bda56591e8e.png)

Reply:
not sure why tests are compiling